components/layout_2020/display_list/mod.rs

```diff
@@ -256,14 +256,49 @@ impl<'a> BuilderForBoxFragment<'a> {
 
         self.build_background(builder);
         self.build_border(builder);
-        let content_rect = self
-            .fragment
-            .content_rect
-            .to_physical(self.fragment.style.writing_mode, self.containing_block)
-            .translate(self.containing_block.origin.to_vector());
-        for child in &self.fragment.children {
-            child.build_display_list(builder, &content_rect)
-        }
+
+        builder.clipping_and_scrolling_scope(|builder| {
+            let overflow_x = self.fragment.style.get_box().overflow_x;
+            let overflow_y = self.fragment.style.get_box().overflow_y;
+            let original_scroll_and_clip_info = builder.current_space_and_clip;
+            if overflow_x != ComputedOverflow::Visible || overflow_y != ComputedOverflow::Visible {
```

Comment:
This seems to treat all values other than `visible` the same. Landing things incrementally is fine, but what’s the plan for fixing that later?

Reply:
The only difference here, I think, is with Overflow::Auto. As far as correctness goes, I think the unnecessary creation of the scroll frame doesn't affect it. Overflow::Hidden is handled below.